crates/sui-framework/packages/deepbook/sources/custodian.move

```diff
@@ -49,109 +46,118 @@ module deepbook::custodian {
         object::uid_to_inner(&account_cap.id)
     }
 
-    public(friend) fun new<T>(ctx: &mut TxContext): Custodian<T> {
-        Custodian<T> {
+    public fun new<App: drop, T>(_app: App, ctx: &mut TxContext): Custodian<App, T> {
```

Comment:
I think one important change in using a witness instead of a `friend` is that we can change the type signature of `friend` function (or delete it) in an upgrade, but of course we cannot do that with a public function.

IMO retaining this flexibility is useful--what do you think @damirka?

Reply:
@sblackshear I went ahead and changed the contents of the PR to have more friends.